gvm/protocols/gmpv9/types.py

```diff
@@ -0,0 +1,277 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from enum import Enum
+
+from typing import Optional
+
+from gvm.errors import InvalidArgument
+
+from gvm.protocols.gmpv8.types import (
+    AlertCondition,
+    AlertEvent,
+    AlertMethod,
+    AliveTest,
+    AssetType,
+    CredentialFormat,
+    CredentialType,
+    FeedType,
+    HostsOrdering,
+    InfoType,
+    PermissionSubjectType,
+    PortRangeType,
+    ScannerType,
+    SeverityLevel,
+    SnmpAuthAlgorithm,
+    SnmpPrivacyAlgorithm,
+    TicketStatus,
+    TimeUnit,
+    get_alert_condition_from_string,
+    get_alert_event_from_string,
+    get_alert_method_from_string,
+    get_alive_test_from_string,
+    get_asset_type_from_string,
+    get_credential_format_from_string,
+    get_credential_type_from_string,
+    get_feed_type_from_string,
+    get_hosts_ordering_from_string,
+    get_info_type_from_string,
+    get_permission_subject_type_from_string,
+    get_port_range_type_from_string,
+    get_severity_level_from_string,
+    get_scanner_type_from_string,
+    get_snmp_auth_algorithm_from_string,
+    get_snmp_privacy_algorithm_from_string,
+    get_ticket_status_from_string,
+)
+
+
+__all__ = [
+    "AlertCondition",
+    "AlertEvent",
+    "AlertMethod",
+    "AliveTest",
+    "AssetType",
+    "CredentialFormat",
+    "CredentialType",
+    "EntityType",
+    "FeedType",
+    "FilterType",
+    "HostsOrdering",
+    "InfoType",
+    "PermissionSubjectType",
+    "PortRangeType",
+    "ScannerType",
+    "SeverityLevel",
+    "SnmpAuthAlgorithm",
+    "SnmpPrivacyAlgorithm",
+    "TicketStatus",
+    "TimeUnit",
+    "get_alert_condition_from_string",
+    "get_alert_event_from_string",
+    "get_alert_method_from_string",
+    "get_alive_test_from_string",
+    "get_asset_type_from_string",
+    "get_credential_format_from_string",
+    "get_credential_type_from_string",
+    "get_entity_type_from_string",
+    "get_feed_type_from_string",
+    "get_filter_type_from_string",
+    "get_hosts_ordering_from_string",
+    "get_info_type_from_string",
+    "get_permission_subject_type_from_string",
+    "get_port_range_type_from_string",
+    "get_scanner_type_from_string",
+    "get_severity_level_from_string",
+    "get_snmp_auth_algorithm_from_string",
+    "get_snmp_privacy_algorithm_from_string",
+    "get_ticket_status_from_string",
+]
+
+
+class EntityType(Enum):
+    """ Enum for entity types """
+
+    AGENT = "note"
+    ALERT = "alert"
+    ASSET = "asset"
+    AUDIT = "audit"
+    CERT_BUND_ADV = "cert_bund_adv"
+    CPE = "cpe"
+    CREDENTIAL = "credential"
+    CVE = "cve"
+    DFN_CERT_ADV = "dfn_cert_adv"
+    FILTER = "filter"
+    GROUP = "group"
+    HOST = "host"
+    INFO = "info"
+    NOTE = "note"
+    NVT = "nvt"
+    OPERATING_SYSTEM = "os"
+    OVALDEF = "ovaldef"
+    OVERRIDE = "override"
+    PERMISSION = "permission"
+    POLICY_CONFIG = "policy"
```

Comment:
Uhhh you spotted another quirk in the current policy/audit/scanconfig/task implementation. Could you check (or also talk to @sarahd93) if audit and policy can be used as entity type e.g. as resource_type for a tag. We had many many discussions about theses topics but I am not sure anymore about the outcome.

Nevertheless it should be POLICY instead of POLICY_CONFIG to be consistent in the naming.
```suggestion
    POLICY = "policy"
```

Reply:
This would be ok for me. I just took the `SCAN_CONFIG` as reference and thought this would be the consistent solution. ;)

Reply:
According to @sarahd93 audit and policy can't be used for tags or similar.
So I will remove them from `EntityType` and `FilterType` again.